decidim-core/app/controllers/decidim/notifications_settings_controller.rb

```diff
@@ -10,6 +10,27 @@ def show
       @notifications_settings = form(NotificationsSettingsForm).from_model(current_user)
     end
 
+    def unsubscribe
```

Comment:
Not sure this logic should belong in `NotificationsController`. Couldn't we reuse the public `Newsletters` controller? This way all the related code lives in the same place.

Reply:
There is no problem to reuse `NewsletterController`. I thought it was appropriate to place it in the `NotificationSettingsController` as it was user's notifications configuration. If you think it's better to put it on `NewslettersController`, we can change it. what we do?

Reply:
Hmmm, I don't have a *really* strong opinion about that. But I feel like a future developer trying to modify this behavior would looks for the logic in `NewslettersController` as it dosn't necessarily share any logic with *notification settings*.

Also, the process of unsubscribing should not require logging in by nature (and actually, I believe that by law as well), and making it public inside `notifications_settings_controller.rb` comes with some ad-hoc extra work.

Related to that, I think the method right now requires signing in. Since we're sending a signed UID, ad-hoc for the `unsubscribe_user` case scenario, I'd use that to identify the user. Forging other user's identities isn't possible (because it's encrypted), and the risk of someone else getting ahold of your UID seems negligible. And in the worst case, the only consequence would be getting you unsubscribed without you noticing it.

If we wanted to prevent that from happening, maybe we could be using an encrypted string that includes a timestamp so we can make that token invalid after two weeks (or a month). Something like:

```ruby
ActiveSupport::MessageEncryptor.new(
  Rails.application.secrets.secret_key_base, 
  "#{user-id}-#{Time.now.to_i}"
)
```

What do you think?

Reply:
OKey. 

We will move to `NewslettersController`. 

It will not be necessary to log in to cancel the subscription and we will use an encrypted string that will be invalid after two weeks or a month. 

Should developers configure the time after the string will be invalid?

Reply:
I don't think we need that to be configurable. Let's put a sensible default (1 month? Two weeks?) and see if anyone demands it.